objectivec/src/error_utils.h

```diff
@@ -10,4 +10,22 @@ NS_ASSUME_NONNULL_BEGIN
 void ORTSaveCodeAndDescriptionToError(int code, const char* description, NSError** error);
 void ORTSaveExceptionToError(const Ort::Exception& e, NSError** error);
 
+// API implementation wrapper macros to handle ORT C++ API exceptions
+// clang-format off
+#define ORT_OBJC_API_IMPL_BEGIN \
+  try {
+
+#define ORT_OBJC_API_IMPL_END(error, failure_return_value) \
+  } catch (const Ort::Exception& e) {                      \
+    ORTSaveExceptionToError(e, (error));                   \
+    return (failure_return_value);                         \
+  }
+// clang-format on
```

Comment:
Does this work in a minimal build with exceptions disabled? May need to use the macros for ORT_TRY/ORT_CATCH etc. 

Also can we leave the '{' and '}' out of the macro so it doesn't hide the creation of a new scope in the code?

Reply:
I think that if we support disabling exceptions while building Objective-C API code, we should use the ORT C API instead. That way we could get error return values instead of program termination.

Reply:
ok. if it's a separate library it should be fine. 

should you check in the cmake file for the objective-c api that onnxruntime_BUILD_SHARED_LIB is set (otherwise the shared lib won't be built)? may just fail when it tried to add the `onnxruntime` as a dependency but a nice error message would be better.

Reply:
adding a check for the shared lib build is a good idea. might be reworking the build files soon for the package, but will add it if we continue to use cmake to build this.